### DISCUSSION TITLE
wiringPi, Zero 2W, Buster?

### DISCUSSION THREAD

**Discussion by ghost** [WiringPi/WiringPi#222] [General]
Tried to install wiringPi on a Zero 2W running Buster. Ran into some issues, but don't want to appear unappreciative or critical. Used wiringPi previously on a Zero W, with much success.

Latest version of wiringPi for Node Red *appears* to install a version of wiringPi that won't work on my Zero 2W. With wiringPi 3.2 for c installed, gpio -v reports Zero 2W compatibility, but after installing the Node Red variant, it seems to 'expect' different hardware, and bombs-out. 

Muchly appreciative for access to Pi GPIO provided. Good work!

**Comment by mwallner** [comment]
Hi @Tressie246 thanks for the feedback.
I'd like to get a little more detail in order to be able to reproduce the issue / improve the current situation. 

=> could you provide some logs of "wiringPi bombing out" - also, did you try to use `gpio` to toggle pins and test if it actually works?

I do also use [Node Red](https://nodered.org/docs/getting-started/raspberrypi) on a couple of Raspberry boards hooked up to [Home Assistant](https://www.home-assistant.io/), unfortunately I do not currently own a Zero 2W that I could do some testing on. (just ordered one, should be around early next week :smile: )

**Reply by ghost** [reply]
Hi @mwallner,

Awesome that you could reply. 

Yes, you'll love the Zero 2W. Loads of CPU. Aside from web surfing, it does a passable desktop for development.

What I can tell you, and that you may soon experience for yourself, is that wiringPi-Node appears hardware incompatible with the Zero 2W. 

```
gpio blink 0
Unable to determine hardware version. I see: Hardware	: BCM2835
,
 - expecting BCM2708 or BCM2709. Please report this to projects@drogon.net
```

Unfortunately, gpio gets no further than the above, and returns without accessing the hardware. As well, gpio -v will reports an earlier version of wiringPi, to the one I had previously installed, then returns a similar error message to the above. I had wiringPi version 3.2 installed, and it appears version 2.25 may have been installed over it.

```
gpio -v
gpio version: 2.25
Copyright (c) 2012-2015 Gordon Henderson
This is free software with ABSOLUTELY NO WARRANTY.
For details type: gpio -warranty

Unable to determine hardware version. I see: Hardware	: BCM2835
,
 - expecting BCM2708 or BCM2709. Please report this to projects@drogon.net
```

Thanks again for the interest. Id really like to employ wiringPi-Node, and am stoked with the possibility of a Zero 2W revision.

Regards,
@Tressie246

**Comment by seamusdemora** [comment]
@Tressie246 
> Tried to install wiringPi on a Zero 2W running Buster. 

Just out of curiosity, where did you find a 'buster' image to install on a Zero 2W?

**Reply by ghost** [reply]
https://downloads.raspberrypi.org/raspios_armhf/images/raspios_armhf-2021-05-28/2021-05-07-raspios-buster-armhf.zip

Seems to work fine. Had to increase the swapfile size some, but after that, no worries.
